File: Data Structures & Algorithms/interleaving-string/submission-0.cs
public class Solution {
    public bool IsInterleave(string s1, string s2, string s3) {
        return DFS(0, 0, 0);

bool DFS(int i, int j, int k)
{
    if (k == s3.Length) return i == s1.Length && j == s2.Length;

    if (i < s1.Length && s1[i] == s3[k])
    {
        if (DFS(i + 1, j, k + 1)) return true;
    }

    if (j < s2.Length && s2[j] == s3[i + j])
    {
        if (DFS(i, j + 1, k + 1)) return true;
    }

    return false;
}
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
         List<List<Integer>> nestedList = new ArrayList<>();
        dfs(root, 0, nestedList);
        return nestedList;
    }

     private void dfs(TreeNode node, int depth, List<List<Integer>> nestedList) {
        if (node == null) {
            return;
        }

        if (nestedList.size() == depth) {
            nestedList.add(new ArrayList<>());
        }

        nestedList.get(depth).add(node.val);
        dfs(node.left, depth +1, nestedList);
        dfs(node.right, depth +1, nestedList);
     }
}
